src/ServiceInsight/Search/SearchBarViewModel.cs

```diff
@@ -52,22 +51,22 @@ protected override void OnActivate()
 
         public void GoToFirstPage()
         {
-            Parent.RefreshMessages(SelectedEndpoint, 1, SearchQuery);
+            Parent.RefreshMessages(FirstLink, 1);
```

Comment:
Why do we still need the current page? Isn't the First/Prev/Last/Next link enough regardless of which page we're in? Same goes for usages below.